dwst/scripts/particles.js

```diff
@@ -20,7 +20,32 @@ const specialChars = [
   '\\',
 ];
 
-export class InvalidParticles extends Error { }
+export class InvalidParticles extends Error {
+  constructor(expression, expected, expectedStringLength, remainder) {
+    super();
+    this.expression = expression;
+    this.expected = expected;
+    // this.got = got;
+    this.remainder = remainder;
```

Comment:
We use names like `this._remainder` to indicate private members.

Reply:
is it a private member?

Reply:
is it used outside the class?